src/string_search.h

```diff
@@ -87,12 +87,12 @@ class StringSearchBase {
   static const int kBMMinPatternLength = 8;
 
   // Store for the BoyerMoore(Horspool) bad char shift table.
-  static int kBadCharShiftTable[kUC16AlphabetSize];
+  int kBadCharShiftTable[kUC16AlphabetSize];
   // Store for the BoyerMoore good suffix shift table.
-  static int kGoodSuffixShiftTable[kBMMaxShift + 1];
+  int kGoodSuffixShiftTable[kBMMaxShift + 1];
   // Table used temporarily while building the BoyerMoore good suffix
   // shift table.
-  static int kSuffixTable[kBMMaxShift + 1];
+  int kSuffixTable[kBMMaxShift + 1];
```

Comment:
I'd rename them to `suffix_table_`, etc., and inline their uses (i.e., get rid of the getters); that makes the code a little safer because it won't drop the array boundaries like it does now.

Reply:
Done! We’re still doing weird things with the array boundaries because of the `- start_` computation which I’m not totally understanding, to be honest, but I don’t see anything speaking against inlining these.